inc/class-book.php

```diff
@@ -109,7 +109,14 @@ static function getBookInformation( $id = null, $contributors_as_string = true,
 		if ( static::useCache() ) {
 			$cached_book_information = wp_cache_get( $cache_id, 'pb' );
 			if ( $cached_book_information ) {
-				if ( ! $read_contributors_from_cache ) {
+				$contributors_cached_as_string = true;
+				foreach ( $contributors->valid as $contributor_type ) {
+					if ( isset( $cached_book_information[ $contributor_type ] ) ) {
+						$contributors_cached_as_string = is_string( $cached_book_information[ $contributor_type ] );
+						break;
+					}
+				}
+				if ( ! $read_contributors_from_cache || $contributors_cached_as_string !== $contributors_as_string ) {
```

Comment:
We need to make sure the format we are requesting as a parameter is the same format we stored in the cache. If those are different, we'll read from the database anyway to return the requested format.